Editor documentation lists non-existant Pebble filters

### Describe the issue

The documentation that shows up in the sidebar in the flow editor shows some filters that don't seem to exist - trying to use them fails like below:
```
2024-10-13 04:24:21.979

io.pebbletemplates.pebble.error.PebbleException: Filter [md5] does not exist. (Hello World! {{ "FooBar" | md5 }}:1)
2024-10-13 04:24:21.979

io.kestra.core.exceptions.IllegalVariableEvaluationException: io.pebbletemplates.pebble.error.PebbleException: Filter [md5] does not exist. (Hello World! {{ "FooBar" | md5 }}:1)
    at io.kestra.core.runners.VariableRenderer.properPebbleException(VariableRenderer.java:64)
    at io.kestra.core.runners.VariableRenderer.renderOnce(VariableRenderer.java:120)
    at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:91)
    at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:76)
    at io.kestra.core.runners.VariableRenderer.render(VariableRenderer.java:68)
    at io.kestra.core.runners.DefaultRunContext.render(DefaultRunContext.java:186)
    at io.kestra.plugin.core.log.Log.run(Log.java:80) at io.kestra.plugin.core.log.Log.run(Log.java:21)
    at io.kestra.core.runners.WorkerTaskThread.doRun(WorkerTaskThread.java:76)
    at io.kestra.core.runners.AbstractWorkerThread.run(AbstractWorkerThread.java:57)
Caused by: io.pebbletemplates.pebble.error.PebbleException: Filter [md5] does not exist. (Hello World! {{ "FooBar" | md5 }}:1)
    at io.pebbletemplates.pebble.node.expression.FilterExpression.evaluate(FilterExpression.java:49)
    at io.pebbletemplates.pebble.node.PrintNode.render(PrintNode.java:37)
    at io.pebbletemplates.pebble.node.BodyNode.render(BodyNode.java:44)
    at io.pebbletemplates.pebble.node.RootNode.render(RootNode.java:31)
    at io.pebbletemplates.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:157)
    at io.pebbletemplates.pebble.template.PebbleTemplateImpl.evaluate(PebbleTemplateImpl.java:96)
    at io.kestra.core.runners.VariableRenderer.renderOnce(VariableRenderer.java:114) ... 8 more
```

![](https://p.cp3.es/ss1728808081-22c655.png)

I tested md5, sha1, and truncate and they fail - there's probably a few more in there.


### Environment

- Kestra Version: 0.19.2 - 4338437 Tue, Oct 8, 2024 9:52 AM
- Operating System (OS/Docker/Kubernetes): Docker (kestra/kestra:latest)
- Java Version (if you don't run kestra in Docker):

--- a/core/src/main/java/io/kestra/core/runners/pebble/Extension.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/Extension.java
@@ -93,6 +93,9 @@ public class Extension extends AbstractExtension {
         filters.put("endsWith", new EndsWithFilter());
         filters.put("values", new ValuesFilter());
         filters.put("toIon", new ToIonFilter());
+        filters.put("sha1", new Sha1Filter());
+        filters.put("sha512", new Sha512Filter());
+        filters.put("md5", new Md5Filter());
         return filters;
     }
 

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Md5Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Md5Filter.java
@@ -1,0 +1,61 @@
+package io.kestra.core.runners.pebble.filters;
+
+
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Filter;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class implements the 'sha256' filter.
+ *
+ * @author Silviu Vergoti
+ */
+public class Md5Filter implements Filter {
+
+
+    @Override
+    public List<String> getArgumentNames() {
+        return null;
+    }
+
+    @Override
+    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
+                        EvaluationContext context, int lineNumber) throws PebbleException {
+        if (input == null) {
+            return null;
+        }
+
+        if (input instanceof String) {
+            MessageDigest digest = null;
+            byte[] encodedHash = null;
+            try {
+                digest = MessageDigest.getInstance("MD5");
+                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
+            } catch (Exception e) {
+                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
+            }
+            return bytesToHex(encodedHash);
+        } else {
+            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
+        }
+    }
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuilder hexString = new StringBuilder(2 * bytes.length);
+        for (int i = 0; i < bytes.length; i++) {
+            String hex = Integer.toHexString(0xff & bytes[i]);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
+
+}

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Md5Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Md5Filter.java
@@ -16,46 +16,9 @@ import java.util.Map;
  *
  * @author Silviu Vergoti
  */
-public class Md5Filter implements Filter {
+public class Md5Filter extends ShaBaseFilter {
 
-
-    @Override
-    public List<String> getArgumentNames() {
-        return null;
+    public Md5Filter() {
+        super("MD5");
     }
-
-    @Override
-    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
-                        EvaluationContext context, int lineNumber) throws PebbleException {
-        if (input == null) {
-            return null;
-        }
-
-        if (input instanceof String) {
-            MessageDigest digest = null;
-            byte[] encodedHash = null;
-            try {
-                digest = MessageDigest.getInstance("MD5");
-                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
-            } catch (Exception e) {
-                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
-            }
-            return bytesToHex(encodedHash);
-        } else {
-            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
-        }
-    }
-
-    private static String bytesToHex(byte[] bytes) {
-        StringBuilder hexString = new StringBuilder(2 * bytes.length);
-        for (int i = 0; i < bytes.length; i++) {
-            String hex = Integer.toHexString(0xff & bytes[i]);
-            if (hex.length() == 1) {
-                hexString.append('0');
-            }
-            hexString.append(hex);
-        }
-        return hexString.toString();
-    }
-
 }

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha1Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha1Filter.java
@@ -1,0 +1,61 @@
+package io.kestra.core.runners.pebble.filters;
+
+
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Filter;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class implements the 'sha256' filter.
+ *
+ * @author Silviu Vergoti
+ */
+public class Sha1Filter implements Filter {
+
+
+    @Override
+    public List<String> getArgumentNames() {
+        return null;
+    }
+
+    @Override
+    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
+                        EvaluationContext context, int lineNumber) throws PebbleException {
+        if (input == null) {
+            return null;
+        }
+
+        if (input instanceof String) {
+            MessageDigest digest = null;
+            byte[] encodedHash = null;
+            try {
+                digest = MessageDigest.getInstance("SHA-1");
+                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
+            } catch (Exception e) {
+                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
+            }
+            return bytesToHex(encodedHash);
+        } else {
+            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
+        }
+    }
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuilder hexString = new StringBuilder(2 * bytes.length);
+        for (int i = 0; i < bytes.length; i++) {
+            String hex = Integer.toHexString(0xff & bytes[i]);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
+
+}

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha1Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha1Filter.java
@@ -16,46 +16,9 @@ import java.util.Map;
  *
  * @author Silviu Vergoti
  */
-public class Sha1Filter implements Filter {
+public class Sha1Filter extends ShaBaseFilter {
 
-
-    @Override
-    public List<String> getArgumentNames() {
-        return null;
+    public Sha1Filter() {
+        super("SHA-1");
     }
-
-    @Override
-    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
-                        EvaluationContext context, int lineNumber) throws PebbleException {
-        if (input == null) {
-            return null;
-        }
-
-        if (input instanceof String) {
-            MessageDigest digest = null;
-            byte[] encodedHash = null;
-            try {
-                digest = MessageDigest.getInstance("SHA-1");
-                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
-            } catch (Exception e) {
-                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
-            }
-            return bytesToHex(encodedHash);
-        } else {
-            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
-        }
-    }
-
-    private static String bytesToHex(byte[] bytes) {
-        StringBuilder hexString = new StringBuilder(2 * bytes.length);
-        for (int i = 0; i < bytes.length; i++) {
-            String hex = Integer.toHexString(0xff & bytes[i]);
-            if (hex.length() == 1) {
-                hexString.append('0');
-            }
-            hexString.append(hex);
-        }
-        return hexString.toString();
-    }
-
 }

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha512Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha512Filter.java
@@ -1,0 +1,61 @@
+package io.kestra.core.runners.pebble.filters;
+
+
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Filter;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class implements the 'sha256' filter.
+ *
+ * @author Silviu Vergoti
+ */
+public class Sha512Filter implements Filter {
+
+
+    @Override
+    public List<String> getArgumentNames() {
+        return null;
+    }
+
+    @Override
+    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
+                        EvaluationContext context, int lineNumber) throws PebbleException {
+        if (input == null) {
+            return null;
+        }
+
+        if (input instanceof String) {
+            MessageDigest digest = null;
+            byte[] encodedHash = null;
+            try {
+                digest = MessageDigest.getInstance("SHA-512");
+                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
+            } catch (Exception e) {
+                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
+            }
+            return bytesToHex(encodedHash);
+        } else {
+            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
+        }
+    }
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuilder hexString = new StringBuilder(2 * bytes.length);
+        for (int i = 0; i < bytes.length; i++) {
+            String hex = Integer.toHexString(0xff & bytes[i]);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
+
+}

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha512Filter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/Sha512Filter.java
@@ -16,46 +16,9 @@ import java.util.Map;
  *
  * @author Silviu Vergoti
  */
-public class Sha512Filter implements Filter {
+public class Sha512Filter extends ShaBaseFilter {
 
-
-    @Override
-    public List<String> getArgumentNames() {
-        return null;
+    public Sha512Filter() {
+        super("SHA-512");
     }
-
-    @Override
-    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
-                        EvaluationContext context, int lineNumber) throws PebbleException {
-        if (input == null) {
-            return null;
-        }
-
-        if (input instanceof String) {
-            MessageDigest digest = null;
-            byte[] encodedHash = null;
-            try {
-                digest = MessageDigest.getInstance("SHA-512");
-                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
-            } catch (Exception e) {
-                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
-            }
-            return bytesToHex(encodedHash);
-        } else {
-            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
-        }
-    }
-
-    private static String bytesToHex(byte[] bytes) {
-        StringBuilder hexString = new StringBuilder(2 * bytes.length);
-        for (int i = 0; i < bytes.length; i++) {
-            String hex = Integer.toHexString(0xff & bytes[i]);
-            if (hex.length() == 1) {
-                hexString.append('0');
-            }
-            hexString.append(hex);
-        }
-        return hexString.toString();
-    }
-
 }

--- a/core/src/main/java/io/kestra/core/runners/pebble/filters/ShaBaseFilter.java
+++ b/core/src/main/java/io/kestra/core/runners/pebble/filters/ShaBaseFilter.java
@@ -1,0 +1,63 @@
+package io.kestra.core.runners.pebble.filters;
+
+import io.pebbletemplates.pebble.error.PebbleException;
+import io.pebbletemplates.pebble.extension.Filter;
+import io.pebbletemplates.pebble.template.EvaluationContext;
+import io.pebbletemplates.pebble.template.PebbleTemplate;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.List;
+import java.util.Map;
+
+
+public class ShaBaseFilter implements Filter {
+
+
+    private String Algorithm;
+
+
+    public ShaBaseFilter(String algorithm) {
+        Algorithm = algorithm;
+    }
+
+    @Override
+    public List<String> getArgumentNames() {
+        return null;
+    }
+
+    @Override
+    public Object apply(Object input, Map<String, Object> args, PebbleTemplate self,
+                        EvaluationContext context, int lineNumber) throws PebbleException {
+        if (input == null) {
+            return null;
+        }
+
+        if (input instanceof String) {
+            MessageDigest digest = null;
+            byte[] encodedHash = null;
+            try {
+                digest = MessageDigest.getInstance(Algorithm);
+                encodedHash = digest.digest(((String) input).getBytes(StandardCharsets.UTF_8));
+            } catch (Exception e) {
+                throw new PebbleException(e, "Hashing exception encountered\n", lineNumber, self.getName());
+            }
+            return bytesToHex(encodedHash);
+        } else {
+            throw new PebbleException(null, "Need a string to hash\n", lineNumber, self.getName());
+        }
+    }
+
+    private static String bytesToHex(byte[] bytes) {
+        StringBuilder hexString = new StringBuilder(2 * bytes.length);
+        for (int i = 0; i < bytes.length; i++) {
+            String hex = Integer.toHexString(0xff & bytes[i]);
+            if (hex.length() == 1) {
+                hexString.append('0');
+            }
+            hexString.append(hex);
+        }
+        return hexString.toString();
+    }
+
+}

--- a/core/src/test/java/io/kestra/core/runners/pebble/filters/Md5FilterTest.java
+++ b/core/src/test/java/io/kestra/core/runners/pebble/filters/Md5FilterTest.java
@@ -1,0 +1,25 @@
+package io.kestra.core.runners.pebble.filters;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.runners.VariableRenderer;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@KestraTest
+public class Md5FilterTest {
+    @Inject
+    VariableRenderer variableRenderer;
+
+    @Test
+    void out() throws IllegalVariableEvaluationException {
+        String render = variableRenderer.render("{{ \"hello\" | md5 }}", Map.of());
+
+        assertThat(render.equals("hello"), is(false));
+    }
+}

--- a/core/src/test/java/io/kestra/core/runners/pebble/filters/Sha1FilterTest.java
+++ b/core/src/test/java/io/kestra/core/runners/pebble/filters/Sha1FilterTest.java
@@ -1,0 +1,25 @@
+package io.kestra.core.runners.pebble.filters;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.runners.VariableRenderer;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@KestraTest
+public class Sha1FilterTest {
+    @Inject
+    VariableRenderer variableRenderer;
+
+    @Test
+    void out() throws IllegalVariableEvaluationException {
+        String render = variableRenderer.render("{{ \"hello\" | sha1 }}", Map.of());
+
+        assertThat(render.equals("hello"), is(false));
+    }
+}

--- a/core/src/test/java/io/kestra/core/runners/pebble/filters/Sha512FilterTest.java
+++ b/core/src/test/java/io/kestra/core/runners/pebble/filters/Sha512FilterTest.java
@@ -1,0 +1,25 @@
+package io.kestra.core.runners.pebble.filters;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.runners.VariableRenderer;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@KestraTest
+public class Sha512FilterTest {
+    @Inject
+    VariableRenderer variableRenderer;
+
+    @Test
+    void out() throws IllegalVariableEvaluationException {
+        String render = variableRenderer.render("{{ \"hello\" | sha512 }}", Map.of());
+
+        assertThat(render.equals("hello"), is(false));
+    }
+}

